### PR TITLE
Feature/save scene

### PR DESCRIPTION
**Pull request by Schneegans**
See https://github.com/cosmoscout/cosmoscout-vr/pull/128.

### CODE REVIEW

File: src/Plugin.cpp [hunk 1]
```diff
@@ -93,13 +78,68 @@ void Plugin::deInit() {
...
+  mAllSettings->onSave().disconnect(mOnSaveConnection);
 
   logger().info("Unloading done.");
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+void Plugin::onLoad() {
+  // Read settings from JSON.
+  from_json(mAllSettings->mPlugins.at("csp-rings"), mPluginSettings);
```
Comment (JonasGilg) [inline src/Plugin.cpp:91]: I'm not sure if this is the correct code, but it looks cleaner with an assignment in my opinion.

```suggestion
  mPluginSettings = mAllSettings->mPlugins.at("csp-rings").get<Plugin::Settings>();
```

File: src/Plugin.hpp [hunk 2]
```diff
@@ -27,10 +29,10 @@ class Plugin : public cs::core::PluginBase {
       std::string mTexture;
 
       /// The distance from the planets center to where the rings start in meter.
-      double mInnerRadius;
+      float mInnerRadius{};
 
       /// The distance from the planets center to where the rings end in meter.
-      double mOuterRadius;
+      float mOuterRadius{};
```
Comment (JonasGilg) [inline src/Plugin.hpp:36]: Why use float here? Shouldn't we use doubles for as long as possible? 

Doubles have better precision especially on the scales we are working on, and on x86 architecture they have the same or even better performance than floats. The only downside is memory consumption, which can be ignored in simple cases like this.

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Review by JonasGilg — COMMENTED** [review]
Looks good, just needs the previous issues resolved.

**Review by JonasGilg — COMMENTED** [review]
Looks good. Can you have a look at my other comments?

File: src/Plugin.cpp (hunk 1 above)
Comment (Schneegans) [reply src/Plugin.cpp:91]: This wouldn't work. By passing mPluginSettings as reference, all the onChange signals of the properties get called when they change due to the loading. If we would overwrite the mPluginSettings, all handlers would be disconnected instead.

File: src/Plugin.hpp (hunk 2 above)
Comment (Schneegans) [reply src/Plugin.hpp:36]: In general, I agree with you. But here they are not used for any calculations, they are directly passed to the shader. So we wouldn't get any benefit, on the contrary - we would have to static_cast them every frame.

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**